tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -122,24 +173,28 @@ object ScatterGather {
     * when the partitions are available and the set of Gathers to be performed on this stage of the Scatter.
     */
   private trait SubScatter[Result <: Task] extends Scatter[Result] {
+    // the tasks that create that results
     val tasks    = new ListBuffer[Result]
+    // the sub-scatterers that further map the results
     val subs     = new ListBuffer[SecondarySubScatter[Result, _ <: Task]]
+    // the gatherers that gather the results
```

Comment:
Ditto, starting with "the" implies there must be gathers, but that's not the case.